diff --git a/node_modules/protocol-buffers-encodings/index.js b/node_modules/protocol-buffers-encodings/index.js
index c10cd9c..5d676e5 100644
--- a/node_modules/protocol-buffers-encodings/index.js
+++ b/node_modules/protocol-buffers-encodings/index.js
@@ -1,6 +1,6 @@
 var varint = require('varint')
 var svarint = require('signed-varint')
-
+const Buffer = require('b4a')
 exports.make = encoder
 
 exports.name = function (enc) {
@@ -39,12 +39,12 @@ exports.bytes = encoder(2,
   function encode (val, buffer, offset) {
     var oldOffset = offset
     var len = bufferLength(val)
-
+    buffer = Buffer.from(buffer)
     varint.encode(len, buffer, offset)
     offset += varint.encode.bytes
 
-    if (Buffer.isBuffer(val)) val.copy(buffer, offset)
-    else buffer.write(val, offset, len)
+    if (Buffer.isBuffer(val)) Buffer.copy(val, buffer, offset)
+    else Buffer.write(buffer, val, offset, len)
     offset += len
 
     encode.bytes = offset - oldOffset
@@ -56,7 +56,7 @@ exports.bytes = encoder(2,
     var len = varint.decode(buffer, offset)
     offset += varint.decode.bytes
 
-    var val = buffer.slice(offset, offset + len)
+    var val = Buffer.slice(buffer, offset, offset + len)
     offset += val.length
 
     decode.bytes = offset - oldOffset
@@ -72,11 +72,10 @@ exports.string = encoder(2,
   function encode (val, buffer, offset) {
     var oldOffset = offset
     var len = Buffer.byteLength(val)
-
     varint.encode(len, buffer, offset, 'utf-8')
     offset += varint.encode.bytes
 
-    buffer.write(val, offset, len)
+    Buffer.write(buffer, val, offset, len)
     offset += len
 
     encode.bytes = offset - oldOffset
@@ -88,7 +87,7 @@ exports.string = encoder(2,
     var len = varint.decode(buffer, offset)
     offset += varint.decode.bytes
 
-    var val = buffer.toString('utf-8', offset, offset + len)
+    var val = Buffer.toString(buffer, 'utf-8', offset, offset + len)
     offset += len
 
     decode.bytes = offset - oldOffset
@@ -158,7 +157,7 @@ exports.int64 = encoder(0,
       while (buffer[offset + limit - 1] === 0xff) limit--
       limit = limit || 9
       var subset = Buffer.allocUnsafe(limit)
-      buffer.copy(subset, 0, offset, offset + limit)
+      Buffer.copy(buffer, subset, 0, offset, offset + limit)
       subset[limit - 1] = subset[limit - 1] & 0x7f
       val = -1 * varint.decode(subset, 0)
       decode.bytes = 10
@@ -192,12 +191,12 @@ exports.varint = encoder(0,
 exports.fixed64 =
 exports.sfixed64 = encoder(1,
   function encode (val, buffer, offset) {
-    val.copy(buffer, offset)
+    Buffer.copy(val,buffer, offset)
     encode.bytes = 8
     return buffer
   },
   function decode (buffer, offset) {
-    var val = buffer.slice(offset, offset + 8)
+    var val = Buffer.slice(buffer, offset, offset + 8)
     decode.bytes = 8
     return val
   },
